plugins/syntax/numeral.ml

```diff
@@ -33,50 +33,59 @@ let get_constructors ind =
   Array.to_list
     (Array.mapi (fun j c -> ConstructRef (ind, j + 1)) mc)
 
-let q_z = qualid_of_string "Coq.Numbers.BinNums.Z"
-let q_positive = qualid_of_string "Coq.Numbers.BinNums.positive"
-let q_int = qualid_of_string "Coq.Init.Decimal.int"
-let q_uint = qualid_of_string "Coq.Init.Decimal.uint"
-let q_option = qualid_of_string "Coq.Init.Datatypes.option"
+let qualid_of_ref n =
+  n |> Coqlib.lib_ref |> Nametab.shortest_qualid_of_global Id.Set.empty
+
+let q_option () = qualid_of_ref "core.option.type"
 
 let unsafe_locate_ind q =
   match Nametab.locate q with
   | IndRef i -> i
   | _ -> raise Not_found
 
-let locate_ind q =
-  try unsafe_locate_ind q
-  with Not_found -> Nametab.error_global_not_found q
-
 let locate_z () =
-  try
-    Some { z_ty = unsafe_locate_ind q_z;
-           pos_ty = unsafe_locate_ind q_positive }
-  with Not_found -> None
+  let zn = "num.Z.type" in
+  let pn = "num.pos.type" in
+  if Coqlib.has_ref zn && Coqlib.has_ref pn
+  then
+    let q_z = qualid_of_ref zn in
+    let q_pos = qualid_of_ref pn in
+    Some ({
+        z_ty = unsafe_locate_ind q_z;
+        pos_ty = unsafe_locate_ind q_pos;
+    }, mkRefC q_z)
+  else None
 
 let locate_int () =
-  { uint = locate_ind q_uint;
-    int = locate_ind q_int }
+  let int = "num.int.type" in
+  let uint = "num.uint.type" in
+  if Coqlib.has_ref int && Coqlib.has_ref uint
+  then
+    let q_int = qualid_of_ref int in
+    let q_uint = qualid_of_ref uint in
+    Some ({
+        int = unsafe_locate_ind q_int;
+        uint = unsafe_locate_ind q_uint;
+    }, mkRefC q_int, mkRefC q_uint)
+  else None
 
 let has_type f ty =
   let (sigma, env) = Pfedit.get_current_context () in
   let c = mkCastC (mkRefC f, Glob_term.CastConv ty) in
   try let _ = Constrintern.interp_constr env sigma c in true
   with Pretype_errors.PretypeError _ -> false
 
-let type_error_to f ty loadZ =
+let type_error_to f ty =
   CErrors.user_err
     (pr_qualid f ++ str " should go from Decimal.int to " ++
      pr_qualid ty ++ str " or (option " ++ pr_qualid ty ++ str ")." ++
-     fnl () ++ str "Instead of Decimal.int, the types Decimal.uint or Z could be used" ++
-     (if loadZ then str " (require BinNums first)." else str "."))
+     fnl () ++ str "Instead of Decimal.int, the types Decimal.uint or Z could be used.")
```

Comment:
What is the reason for dropping the parenthetical?  You could instead augment it to say `require BinNums or Register a constant as ... first`.

Either way, please also update the documentation to match the updated error message.